src/gazebo_gps_plugin.cpp

```diff
@@ -286,17 +302,44 @@ void GpsPlugin::OnWorldUpdate(const common::UpdateInfo& /*_info*/)
   gps_bias_.Y() += random_walk_gps_.Y() * dt - gps_bias_.Y() / gps_corellation_time_;
   gps_bias_.Z() += random_walk_gps_.Z() * dt - gps_bias_.Z() / gps_corellation_time_;
 
+
+    
+
+{
+    std::scoped_lock lock(gps_spoof_mutex_);
+    if (spoofing_active_) {
+      //gzdbg << "GPS SPOOFING ACTIVE " << "\n";
+      gzdbg << "GPS SPOOFING Time: " << (current_time - spoof_start_time).Double() <<"\n";
+      // gps_attack_.X() = 1000;
+      // gps_attack_.Y() = 1000;
+
+      gps_attack_.X() = gps_attack_.X() + 0.005;
+      gps_attack_.Y() = gps_attack_.Y() + 0.005;
+    }
+}
+
   // reproject position with noise into geographic coordinates
-  auto pos_with_noise = pos_W_I + noise_gps_pos_ + gps_bias_;
-  auto latlon = reproject(pos_with_noise, lat_home_, lon_home_, alt_home_);
+  auto pos_with_noise = pos_W_I + noise_gps_pos_ + gps_bias_ + gps_attack_;
+  auto latlon         = reproject(pos_with_noise, lat_home_, lon_home_, alt_home_);
+  gzdbg << "Lat: " << latlon.first << "  Lon: " << latlon.second << "\n";
 
+  {
+    std::scoped_lock lock(gps_spoof_mutex_);
+    if ((spoofing_active_) && ((current_time - spoof_start_time).Double() > 32) )
```

Comment:
What is purpose of this part? After spoofing has been active for 32 seconds, the gps coordinates jumps to some static location?